tsl/test/sql/hypercore_trigger.sql

```diff
@@ -0,0 +1,118 @@
+-- This file and its contents are licensed under the Timescale License.
+-- Please see the included NOTICE for copyright information and
+-- LICENSE-TIMESCALE for a copy of the license.
+
+\ir include/setup_hypercore.sql
+
+create table saved_rows (like :chunk1, new_row bool not null, kind text);
+create table count_stmt (inserts int, updates int, deletes int, truncates int);
+
+create function save_insert_row() returns trigger as $$
+begin
+   insert into saved_rows select new.*, true, 'insert';
+   return new;
+end;
+$$ language plpgsql;
+
+create function save_update_row() returns trigger as $$
+begin
+   insert into saved_rows select new.*, true, 'update';
+   insert into saved_rows select old.*, false, 'update';
+   return new;
+end;
+$$ language plpgsql;
+
+create function save_delete_row() returns trigger as $$
+begin
+   insert into saved_rows select old.*, false, 'delete';
+   return new;
+end;
+$$ language plpgsql;
```

Comment:
> I think it can be simplified in only one trigger function and also track `TG_WHEN` and use `TG_OP` to save the proper operation on the `saved_rows` table.

Looking into if we can optimize the test in this manner.

> Another thing is that you're testing only `BEFORE` triggers and what about `AFTER` and `INSTEAD OF` ??

All triggers are executed from PostgreSQL code, and the important part from our implementation perspective is that we are passing reading rows correctly from the TAM and set them properly for NEW and OLD.

Since the BEFORE trigger use both NEW and OLD. If the NEW and OLD are correct for BEFORE triggers, they will be correct for INSTEAD OF and AFTER triggers. If not, there is a bug in PostgreSQL code, not our code, so for that reason, I don't think we need to test with AFTER and INSTEAD OF triggers.

It is also possible that the number of row-level calls is incorrect if we return a vector of decompressed values, so checking the count here is important. If it does not match what you would get with a fully-decompressed table, we have a bug in our code.

Since we have two relations internally, it is theoretically possible that the statement-level triggers are not called the correct number of times, so we need to check that. We should also check that the values are correct for statement-level triggers based on data that arrives from the storage.

I realize that I am missing a test for a transition table, so I will add that.

Reply:
Using a single trigger function for all tests and added tests for transition tables (which right now just generate an error).